tensorflow/python/ops/array_ops.py

```diff
@@ -1608,6 +1608,138 @@ def _DepthToSpaceShape(op):
       [input_shape[0], height, width, new_depth])]
 
 
+def one_hot(indices, depth, on_value=None, off_value=None,
+            axis=None, dtype=dtypes.float32, name=None):
```

Comment:
on_value and off_value and dtype should be None by default.

if on_value and off_value are provided; then use their dtype.  otherwise if dtype is provided, use on_value 1 and off_value 0 for that dtype.  if dtype is tf.string, raise an error because it's not clear what the on_value should be.  if all 3 are provided, make sure the dtype of on_value and off_value match dtype.

update the Args: section below to reflect the changes.

update the Args: section below to reflect this.